Modules/comm.lua

```diff
@@ -997,72 +285,87 @@ function CommDKP.Sync:SendData(prefix, data, target, targetTeam)
-      CommDKP.Sync:SendCommMessage(prefix, _compressedObj, "RAID")
+    elseif prefix == "CommDKPTalents" then
+      CommDKPTalentsSend(prefix, _compressedObj, "GUILD");
       return;
-    end
-  end
-
-  -- officers
-  if IsInGuild() and core.IsOfficer then
-    
-    if prefix == "CommDKPCommand" or prefix == "CommDKPRaidTime" then
-      CommDKP.Sync:SendCommMessage(prefix, _compressedObj, "RAID")
+    elseif prefix == "CommDKPRoles" then
+      CommDKPRolesSend(prefix, _compressedObj, "GUILD");
       return;
-    end
-
-    if prefix == "CommDKPBCastMsg" then
-      CommDKP.Sync:SendCommMessage(prefix, _compressedObj, "RAID") -- changed to raid from guild
+    elseif prefix == "CDKProfileSend" then
+      CommDKPProfileSend(prefix, _compressedObj, "GUILD");
       return;
-    end  
-
-    if (prefix == "CommDKPZSumBank" or prefix == "CommDKPBossLoot" or prefix == "CommDKPBidShare") then    -- Zero Sum bank/loot table/bid table data and bid submissions. Keep to raid.
-      CommDKP.Sync:SendCommMessage(prefix, _compressedObj, "RAID")
+    elseif prefix == "CommDKPBidder" then -- bid submissions. Keep to raid.
+      CommDKPBidderSend(prefix, _compressedObj, "RAID");
       return;
-    end
-
-    if prefix == "CommDKPPreBroad" then
-      if target then
-        CommDKP.Sync:SendCommMessage(prefix, _compressedObj, target);
-      else
-        CommDKP.Sync:SendCommMessage(prefix, _compressedObj, "GUILD");
-      end
+    elseif prefix == "CommDKPBuild" then
+      CommDKP.Sync:SendCommMessage(prefix, _compressedObj, "GUILD");
       return;
     end
 
-    if prefix == "CommDKPAllTabs" or prefix == "CommDKPMerge" then
+    if core.IsOfficer then
+      if prefix == "CommDKPCommand" then
+        CommDKPCommandSend(prefix, _compressedObj, "RAID");
+        return;
+      end
+  
+      if prefix == "CommDKPRaidTime" then
+        CommDKPRaidTimeSend(prefix, _compressedObj, "RAID");
+        return;
+      end
+  
+      if prefix == "CommDKPBCastMsg" then
+        CommDKPCastMsgSend(prefix, _compressedObj, "RAID");
+        return;
+      end  
+  
+      if prefix == "CommDKPZSumBank" then
+        CommDKPZSumBankSend(prefix, _compressedObj, "RAID");
+        return;
+      end  
+  
+      if prefix == "CommDKPBossLoot" then
+        CommDKPBossLootSend(prefix, _compressedObj, "RAID");
+        return;
+      end  
+  
+      if prefix == "CommDKPBidShare" then
+        CommDKPBidShareSend(prefix, _compressedObj, "RAID");
+        return;
+      end  
+  
+      if prefix == "CommDKPPreBroad" then
+        CommDKPPreBroadSend(prefix, _compressedObj, target);
+        return;
+      end
+  
+      if prefix == "CommDKPAllTabs" then
+        FullBroadcastSend(prefix, _compressedObj, target);
+        return;
+      end
+  
+      if prefix == "CommDKPMerge" then
+        TableMergeSend(prefix, _compressedObj, target);
+        return;
+      end
+      
+      -- what is being sent here?

```

Comment:
This appears to be the section that lets an officer manually broadcast their tables. The print lines should probably use CommDKP's internal print function rather than be removed entirely.

Removing the callback function won't mess with the broadcasting in any way? The progress indicator still appears when sending tables?

Reply:
I dont think I removed the callback function, its stored in Sync.lua

Reply:
You don't pass it to SendCommMessage anymore though. It looks like the callback function is used to reset tracking how many bytes have been sent so far. So I think this might make it so that a second broadcast would have an incorrect progress bar. Have you tested broadcasting multiple times with this change?

Reply:
Also I think it would remove the messages in chat when a broadcast is started, and I'm not sure if that's intentional. I think they're good feedback.

Reply:
SendCommMessage  is only sent in FullBroadcast and 2-WeekMerge to display the % window for person performing the broadcast. No other message was using this callback. I didnt remove this callback from the 2 previously mentioned msgs